hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsSelector.java

```diff
@@ -0,0 +1,240 @@
+    DataSourceUtils.checkRequiredProperties(
+        props, Collections.singletonList(CloudObjectsSelector.Config.QUEUE_URL_PROP));
+    this.props = props;
+    this.queueUrl = props.getString(CloudObjectsSelector.Config.QUEUE_URL_PROP);
+    this.longPollWait =
+        (int) props.getOrDefault(CloudObjectsSelector.Config.QUEUE_LONGPOLLWAIT_PROP, 20);
+    this.maxMessageEachBatch =
+        (int) props.getOrDefault(CloudObjectsSelector.Config.QUEUE_MAXMESSAGESEACHBATCH_PROP, 5);
+    this.visibilityTimeout =
+        (int) props.getOrDefault(CloudObjectsSelector.Config.QUEUE_VISIBILITYTIMEOUT_PROP, 30);
+    this.maxMessagesEachRequest = 10;
+  }
+
+  /** Get the file attributes filePath, eventTime and size from JSONObject record. */
+  public static Map<String, Object> getFileAttributesFromRecord(JSONObject record)
+      throws UnsupportedEncodingException {
+
+    Map<String, Object> fileRecord = new HashMap<>();
+    String eventTimeStr = record.getString("eventTime");
+    long eventTime =
+        Date.from(Instant.from(DateTimeFormatter.ISO_INSTANT.parse(eventTimeStr))).getTime();
+
+    JSONObject s3Object = record.getJSONObject("s3").getJSONObject("object");
+    String bucket =
+        URLDecoder.decode(
+            record.getJSONObject("s3").getJSONObject("bucket").getString("name"), "UTF-8");
+    String key = URLDecoder.decode(s3Object.getString("key"), "UTF-8");
+    String filePath = "s3://" + bucket + "/" + key;
+
+    fileRecord.put("eventTime", eventTime);
+    fileRecord.put("fileSize", s3Object.getLong("size"));
+    fileRecord.put("filePath", filePath);
+    return fileRecord;
+  }
+
+  /** Amazon SQS Client Builder. */
+  public static AmazonSQS createAmazonSqsClient() {
+    // ToDO - Update it for handling AWS Client creation but not using default only.
+    return AmazonSQSClientBuilder.defaultClient();
+  }
+
+  /** Get SQS queue attributes. */
+  public static Map<String, String> getSqsQueueAttributes(AmazonSQS sqs, String queueUrl) {
+    GetQueueAttributesResult queueAttributesResult =
+        sqs.getQueueAttributes(
+            new GetQueueAttributesRequest(queueUrl)
+                .withAttributeNames("ApproximateNumberOfMessages"));
+    return queueAttributesResult.getAttributes();
+  }
+
+  /** List messages from queue. */
+  public static List<Message> getMessagesToProcess(
+      AmazonSQS sqs,
+      String queueUrl,
+      ReceiveMessageRequest receiveMessageRequest,
+      int maxMessageEachBatch,
+      int maxMessagesEachRequest) {
+    List<Message> messagesToProcess = new ArrayList<>();
+
+    // Get count for available messages
+    Map<String, String> queueAttributesResult = getSqsQueueAttributes(sqs, queueUrl);
+    long approxMessagesAvailable =
+        Long.parseLong(queueAttributesResult.get("ApproximateNumberOfMessages"));
+    log.info("Approx. " + approxMessagesAvailable + " messages available in queue.");
+
+    for (int i = 0;
```

Comment:
Let's add a comment why this loop is needed and why we can't use any collection api for partition.